icws-panel/js/AppCtrl.js

```diff
@@ -1,30 +1,31 @@
 angular.module('IcwsPanel').controller('AppCtrl', ['$scope', '$window', function AppCtrl($scope, $window){
-    var ctrl = this;
+    let ctrl = this;
 
     this.requestEntries = {};
-    this.communicationEntries = [];
+    this.displayedEntries = this.communicationEntries = [];
```

Comment:
This line is setting displayedEntries and communicationEntries to the _same_ object.  When you push to one, the other will reflect that change.  I'm not sure that is what you want.

Reply:
Can we change the names to "filteredEntries" and "allEntries" just to make it a little bit more clear what each is?